pkg/cmd/samples/create.go

```diff
@@ -59,125 +54,161 @@ func (cc *CreateCmd) runCreateCmd(cmd *cobra.Command, args []string) error {
 		return nil
 	}
 
-	sample := samples.Samples{
-		Config: cc.cfg,
-		Fs:     afero.NewOsFs(),
-		Git:    gitpkg.Operations{},
-	}
-
-	if _, ok := sample.GetSamples("create")[args[0]]; !ok {
-		errorMessage := fmt.Sprintf(`The sample provided is not currently supported by the CLI: %s
-To see supported samples, run 'stripe samples list'`, args[0])
-		return fmt.Errorf(errorMessage)
-	}
-
 	selectedSample := args[0]
-	color := ansi.Color(os.Stdout)
-
 	destination := selectedSample
 	if len(args) > 1 {
 		destination = args[1]
 	}
 
-	exists, _ := afero.DirExists(sample.Fs, destination)
-	if exists {
-		return fmt.Errorf("Path already exists for: %s", destination)
-	}
-
+	color := ansi.Color(os.Stdout)
 	spinner := ansi.StartNewSpinner(fmt.Sprintf("Downloading %s", selectedSample), os.Stdout)
 
-	if cc.forceRefresh {
-		err := sample.DeleteCache(selectedSample)
-		if err != nil {
-			logger := log.Logger{
-				Out: os.Stdout,
-			}
-
-			logger.WithFields(log.Fields{
-				"prefix": "samples.create.forceRefresh",
-				"error":  err,
-			}).Debug("Could not clear cache")
-		}
-	}
-
-	// Initialize the selected sample in the local cache directory.
-	// This will either clone or update the specified sample,
-	// depending on whether or not it's. Additionally, this
-	// identifies if the sample has multiple integrations and what
-	// languages it supports.
-	err := sample.Initialize(selectedSample)
+	sampleConfig, err := samples.GetSampleConfig(selectedSample, cc.forceRefresh)
 	if err != nil {
-		switch e := err.Error(); e {
-		case git.NoErrAlreadyUpToDate.Error():
-			// Repo is already up to date. This isn't a program
-			// error to continue as normal
-			break
-		case git.ErrRepositoryAlreadyExists.Error():
-			// If the repository already exists and we don't pull
-			// for some reason, that's fine as we can use the existing
-			// repository
-			break
-		default:
-			ansi.StopSpinner(spinner, "An error occurred.", os.Stdout)
-			return err
-		}
+		ansi.StopSpinner(spinner, "", os.Stdout)
+		return err
 	}
-
 	ansi.StopSpinner(spinner, "", os.Stdout)
 	fmt.Printf("%s %s\n", color.Green("✔"), ansi.Faint("Finished downloading"))
 
 	// Once we've initialized the sample in the local cache
 	// directory, the user needs to select which integration they
 	// want to work with (if selectedSamplelicable) and which language they
 	// want to copy
-	err = sample.SelectOptions()
+	selectedConfig, err := promptSampleConfig(sampleConfig)
 	if err != nil {
 		return err
 	}
 
-	// Setup to intercept ctrl+c
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt)
+	resultChan := make(chan samples.CreationResult)
+
+	go samples.Create(
+		cc.cfg,
+		selectedSample,
+		selectedConfig,
+		destination,
+		cc.forceRefresh,
+		resultChan,
+	)
+
+	for res := range resultChan {
```

Comment:
you might want a `default` case for safety

Reply:
Added a default case, ptal @pepin-stripe

Reply:
did you push it? I don't see it.

Reply:
Yep here it is:
https://github.com/stripe/stripe-cli/pull/652/files#diff-54c4064d2b4ba0c610ea0511c0faa5109f448125b2b307a35ee5c53939c0310aR117-R119

Reply:
wtf it didnt appear before I clicked that link! LGTM 🚢